### PR TITLE
logsumexp handling of edge-cases

### PR DESCRIPTION
**Pull request by rasmushaugaard**
Hi! First of all thanks for this library, which has been useful in a few projects.

I've run into some inconsistencies between the torch_scatter.logsumexp and torch.logsumexp:

1) logsumexp of no elements should default to `-inf`, not `0`.
    This is equivalent to having sum(..., start=0) in non-log space.  
    `torch.logsumexp(torch.tensor([]), dim=0)` returns `-inf`.
2) it should not hide nans:
    `torch.logsumexp(torch.tensor([torch.nan]), dim=0)` returns `nan`.
3) no `eps` should be added. `torch.logsumexp` also doesn't have this.

In the first commit, I've added test cases for empty lists, -inf, inf, very large positive and negative numbers - and added a cartesian product test across the inputs and the various floating point data types.

In the second commit, I've changed the implementation to comply with the new tests.

### Dependency Management

On another note, I ended up spending a few hours setting up the torch/cuda dependencies correctly to build from source.
I ended up using the following commands to install deps, build from source and run the tests:

```bash
cd pytorch_scatter
conda create -n torchscatter python=3.10 pytorch::pytorch pytorch::pytorch-cuda=11.8 nvidia/label/cuda-11.8.0::cuda
conda activate torchscatter
pip install -e ".[test]"
pytest
```

Maybe having this or something similar in the docs (if it's not already there and I missed it) could make it easier for others to contribute.

### CODE REVIEW

**Comment by rasmushaugaard** [comment]
I just had a look at the history, and it seems an issue #368 motivated a change of how nans and infs were treated in #369 and introduced a new issue #407 - which is also the issue that motivated this pull request, which should solve both.

I've added tests and improved numerical stability for the inplace version.

**Comment by rasmushaugaard** [comment]
Hi @rusty1s, Is there something I can do to help with this? :)
